src/foundry/client/pixi/layers/controls/ruler.d.mts

```diff
@@ -81,41 +49,71 @@ declare global {
     labels: PIXI.Container;
 
     /**
-     * Track the current measurement state
-     * @defaultValue `Ruler.STATES.INACTIVE`
+     * The possible Ruler measurement states.
      */
-    protected _state: ValueOf<(typeof Ruler)["STATES"]>;
+    static get STATES(): {
+      INACTIVE: 0;
+      STARTING: 1;
+      MEASURING: 2;
+      MOVING: 3;
+    };
+
+    /**
+     * Is the ruler ready for measure?
+     */
+    static get canMeasure(): boolean;
 
     /**
      * The current destination point at the end of the measurement
-     * @defaultValue `{x: undefined, y: undefined}`
+     * @defaultValue `null`
+     */
+    destination: Point | null;
+
+    /**
+     * The origin point of the measurement, which is the first waypoint.
+     * @defaultValue `null`
+     */
+    get origin(): Point | null;
+
+    /**
+     * This Array tracks individual waypoints along the ruler's measured path.
+     * The first waypoint is always the origin of the route.
+     * @defaultValue `[]`
      */
-    destination: PIXI.Point | null;
+    waypoints: Point[];
 
     /**
      * The array of most recently computed ruler measurement segments
+     * @defaultValue `[]`
+     */
+    segments: Ruler.MeasurementSegment[];
+
+    /**
+     * The measurement history.
      */
-    segments: RulerMeasurementSegment[];
+    get history(): Ruler.MeasurementHistory;
 
     /**
      * The computed total distance of the Ruler.
+     * @defaultValue `0`
      */
     totalDistance: number;
 
     /**
-     * An enumeration of the possible Ruler measurement states.
+     * The computed total cost of the Ruler.
+     * @defaultValue `0`
      */
-    static STATES: {
-      INACTIVE: 0;
-      STARTING: 1;
-      MEASURING: 2;
-      MOVING: 3;
-    };
+    totalCost: number;
 
     /**
-     * Is the ruler ready for measure?
+     * The current state of the Ruler (one of {@link Ruler.STATES}).
      */
-    static get canMeasure(): boolean;
+    get state(): ValueOf<(typeof Ruler)["STATES"]>;
+
+    /**
+     * The current state of the Ruler (one of {@link Ruler.STATES}).
+     */
+    protected _state: ValueOf<(typeof Ruler)["STATES"]>;
```

Comment:
Nit, you can write `typeof Rule.STATES`

Reply:
Does not appear to work in this context due to TS thinking that I'm trying to reference the namespace

Reply:
I see, annoying.